rosidl_generator_py/resource/_msg.py.template

```diff
@@ -98,7 +102,7 @@ class @(spec.base_type.type)(metaclass=Metaclass):
         self.@(field.name) = kwargs.get(
             '@(field.name)', @(spec.base_type.type).@(field.name.upper())__DEFAULT)
 @[    else]@
-@[      if not field.type.is_primitive_type()]@
+@[      if not field.type.is_primitive_type() and (not field.type.is_array or field.type.array_size)]@
```

Comment:
The condition is the same as for the following code which only uses the type in line 115 (`array_size`) and 120 (`not is_array`).